### PR TITLE
Adds initial metrics

### PR DESCRIPTION
**Pull request by michaelgoin**
## Links

* Closes: https://github.com/newrelic/newrelic-node-apollo-server-plugin/issues/36
* Dashboard example leveraging metrics: [a long url](https://staging-one.newrelic.com/launcher/dashboards.launcher?pane=eyJuZXJkbGV0SWQiOiJkYXNoYm9hcmRzLmRhc2hib2FyZCIsImVudGl0eUlkIjoiTVRrd2ZGWkpXbnhFUVZOSVFrOUJVa1I4TVRZeU5qQTAiLCJ1c2VEZWZhdWx0VGltZVJhbmdlIjpmYWxzZSwiaXNUZW1wbGF0ZUVtcHR5IjpmYWxzZX0=&platform[$isFallbackTimeRange]=false&platform[accountId]=190)

## Details

* Operation segment names now also include the deepest path.
* Operation segments now use a final name part of '`<unknown>`' instead of '`<operation unknown>`'
  The additional 'operation' word was redundant now that 'operation' is an earlier part of the name (prior pr).
* Adds initial metrics for operations and field resolve.
  * Operation metrics match 1:1 with segment name
  * Resolve metrics use the field name **NOT** the path for comparison across all usages.
    We may eventually want to generate a path metric too? I figure it is easier to add later than remove something we don't need.
* Adds support metric to track usage/uptake of the module.

Some test files got moved around so they can be leveraged outside the versioned tests.

Support metric will need to be added to visualization tracking.

### CODE REVIEW

File: tests/integration/metrics.test.js [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+  t.test('anonymous query, single level metrics', (t) => {
+    const { helper, serverUrl } = t.context
+
+    const query = `query {
+      hello
+    }`
+
+    helper.agent.on('transactionFinished', () => {
+      const operationPart = `query/${ANON_PLACEHOLDER}/hello`
+      t.metrics([
```
Comment (carlo-808) [inline tests/integration/metrics.test.js:37]: This must be the tap extensions from test-utils 👍

File: tests/integration/metrics.test.js [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+    const invalidQuery = `query {
+      libraries {
+        books {
+          title
+          author {
+            name
+          }
+        }
+      }
+    ` // missing closing }
```
Comment (carlo-808) [inline tests/integration/metrics.test.js:167]: Nice calling out whats invalid about query in comment

File: lib/create-plugin.js [hunk 3]
```diff
@@ -279,4 +283,45 @@ function getDeepestSelectionPath(definition) {
...
 }
 
+function recordResolveSegment(segment, scope) {
+  const duration = segment.getDurationInMillis()
+  const exclusive = segment.getExclusiveDurationInMillis()
+
+  const transaction = segment.transaction
+
+  const attributes = segment.getAttributes()
+  const fieldName = attributes[FIELD_NAME_ATTR]
```
Comment (carlo-808) [inline lib/create-plugin.js:293]: I think this will work with an alias for a field.  minor: add a test with an alias?

**Review by carlo-808 — APPROVED** [review]
I see we are adding metrics for Operation and Resolver timings with some nice helper functions for tests.  👍

File: lib/create-plugin.js (hunk 3 above)
Comment (michaelgoin) [reply lib/create-plugin.js:293]: We don't really have anything considering alias' at all.

Perhaps enter a new issue to explore the impacts of aliases and add tests to the various considerations (transaction naming, segment naming, attributes, metrics).

Comment (carlo-808) [reply lib/create-plugin.js:293]: Sounds good 👍

**Pull request merged**